src/openassetio-python/package/openassetio/constants.py

```diff
@@ -49,64 +44,4 @@
 ## called from multi-threaded C++).
 kField_EntityReferencesMatchPrefix = "entityReferencesMatchPrefix"
```

Comment:
[question] As far as I can tell, this is the only constant that we _require_.  If that's likely to remain the case, then perhaps the entity prefix should have special status, e.g. a `ManagerInterface::entityReferencePrefix() -> std::optional<std::string>` method.

Alternatively...

The entity reference prefix allows us to to help manager plugins simplify, very slightly, their code; as well as optimise away a single virtual function call, at the cost of an `if-else` branch.  

So maybe for such small gains, and assuming this constant is the only one we require, we could get rid of it?

Reply:
Its not for code simplification, its about not acquiring the GIL in a python implementation that has a static prefix. There is no guarantee that this is ever the case (some managers could consume file paths as entity references if they are under a certain filesystem root for example).

A first class method might be a nice way to formalize it if we prefer that. The reason it was in there, is there are also a bunch of other things we used to put in there eg: `Icon`, `Smallcon` etc... that were used by integrations. So I guess we don't want to snowball into too many methods.

Reply:
Ah, that damned GIL.  Makes sense.